BetSystem/Endpoint/SportEventEndpoint.cs

```diff
@@ -2,28 +2,105 @@
+        
+
+        private static IResult GetResultFromSportEvent(int id, [FromServices] ISportEventService service)
         {
-            var result = validator.Validate(sportEventDto);
+            var sportEvent = service.GetById(id);
+            if (sportEvent == null) return Results.NotFound();
+            return Results.Ok(service.GetResultsFromSportEvent(id));
+        }
+
+        private static IResult PostResultToSportEvent(int id, EventResultDto eventResultDto, IValidator<EventResultDto> validator, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+
+            if (sportEvent == null) return Results.NotFound();
+            var result = validator.Validate(eventResultDto);
+
+            if (!result.IsValid) return Results.ValidationProblem(result.ToDictionary());
+            if (service.GetById(id) == null) Results.NotFound(id);
+            if (service.GetById(eventResultDto.TeamId) == null) return Results.BadRequest("Team Not Found");
+
+            service.AddResultToSportEvent(id, eventResultDto);
+
+            return Results.Ok();
+
+        }
+
+        private static IResult GetBetToSportEvent(int id, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+            if (sportEvent == null) return Results.NotFound();
+            return Results.Ok(service.GetBetsFromSportEvent(id));
+        }
+
+        private static IResult PlaceBetOnEvent(int id,IValidator<BetOnEventDto> validator, BetOnEventDto betOnEventDto, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+            if (sportEvent == null) return Results.NotFound();
+            var result = validator.Validate(betOnEventDto);
             if (!result.IsValid)
             {
                 return Results.ValidationProblem(result.ToDictionary());
             }
+            if (service.GetById(betOnEventDto.TeamId) == null) return Results.BadRequest("Team Not Found");
+            if (service.GetById(betOnEventDto.EventId) == null) return Results.BadRequest("Event Not Found");
+
+            service.AddBetToSportEvent( betOnEventDto);
+            return Results.Created($"/BetOnEvent/{betOnEventDto.Id}", betOnEventDto);
+        }
 
+        private static IResult GetTeamsFromSportEvent(int id, [FromServices] ISportEventService service)
+        {
+            var sportEvent = service.GetById(id);
+
+            if (sportEvent == null) return Results.NotFound();
+            return Results.Ok(service.GetTeamsFromSportEvent(id));
+
+
+        }
+
+        public static IResult PostTeamToSportEvent(int id, IdRequestDto idRequestDto, IValidator<IdRequestDto> validator, [FromServices] ISportEventService service, [FromServices] ITeamService teamService)
+        {
+            if (service.GetById(id) == null) return Results.NotFound();
```

Comment:
Question if we just check and later paste id and then probably we will again retrieve event by Id - maybe make sense to pass event object in line 95 - not id?

Reply:
I think not